### PR TITLE
Move siphoning to a tool quality

### LINKED ISSUE #40654: Siphoning fuel requires hose but ignores leather hose
# Describe the bug

When i examine vehicle and i want siphon fuel from tanks, game requires hose and ignores leather hose which i already have. When wield Leather hose and use it manually, it works correctly.

# Steps To Reproduce

1. Craft leather hose.
2. 'e'xamine car
3. 's'iphon

# Expected behavior

I except that leather hose will be used as normal hose.

# Screenshots

https://cdn.discordapp.com/attachments/642349261320880128/711651845465964544/unknown.png

# Versions and configuration

20:35:09.628 INFO : Cataclysm DDA version 0.E-2191-g152f4029ed
20:35:09.629 INFO : [main] C locale set to pl_PL.utf8
20:35:09.629 INFO : [main] C++ locale set to pl_PL.utf8
20:35:09.643 INFO : [options] C locale set to pl_PL.utf8
20:35:09.643 INFO : [options] C++ locale set to pl_PL.utf8
20:35:09.899 WARNING : opendir [./mods/] failed with "No such file or directory".
20:35:15.222 WARNING : opendir [./save/Laura/mods] failed with "No such file or directory".

 - OS: Fedora release 31 (Thirty One)
 - Game Version: Cataclysm DDA version 0.E-2191-g152f4029ed
 - Graphics version: Tiles and Terminal
 - Ingame language: English

### PR DESCRIPTION
**Pull request by Testudo-Cultu**
#### Summary

SUMMARY: Bugfixes "Fix #40654 using ZhilkinSerg's suggestion"

#### Purpose of change

Moves siphoning to a tool quality to make it easier to implement the function in other tools.

#### Describe the solution

It's minor and straightforward. It's possible I did something wrong, but in testing it worked.

#### Describe alternatives you've considered

Obtaining a pet rock and naming it Billy

#### Testing

Tested makeshift and rubber siphons. They both work.

#### Additional context

I basically just did what Zhilkin said to do, so really this is more his fix than mine.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Please also update documentation for this new flag.

**Comment by jbytheway** [comment]
Looks like #40705 was opened at the same time to the same end.

**Comment by Testudo-Cultu** [comment]
> Please also update documentation for this new flag.

Please let me know if I did it correctly and if I need to adjust anything.

**Comment by Night-Pryanik** [comment]
> Please let me know if I did it correctly and if I need to adjust anything.

Oops, sorry. I totally missed the fact that you're adding a new quality, not a new flag. We don't have proper documentation for qualities.

**Comment by Testudo-Cultu** [comment]
> Oops, sorry. I totally missed the fact that you're adding a new quality, not a new flag. We don't have proper documentation for qualities.

Oh. That explains why I was confused, I suppose. I'll fix it.

File: src/veh_interact.cpp [hunk 1]
```diff
@@ -580,7 +581,7 @@ task_reason veh_interact::cant_do( char mode )
                     break;
                 }
             }
-            has_tools = crafting_inv.has_tools( itype_hose, 1 );
+            has_tools = g->u.max_quality( qual_HOSE_SIPHON );
```
Comment (ZhilkinSerg) [inline src/veh_interact.cpp:584]:
```suggestion
            has_tools = g->u.has_quality( qual_HOSE_SIPHON );
```

File: data/json/tool_qualities.json [hunk 2]
```diff
@@ -176,6 +176,12 @@
     "id": "SELF_JACK",
     "name": { "str": "self jacking" }
   },
+  {
+    "type": "tool_quality",
+    "id": "HOSE_SIPHON",
```
Comment (ZhilkinSerg) [inline data/json/tool_qualities.json:181]: I would probably name this quality just `HOSE`.

Comment (Testudo-Cultu) [reply data/json/tool_qualities.json]: Done.

**Comment by Rivet-the-Zombie** [comment]
Swapping out item-specific checks for flags is always good.

**Pull request merged**